### DISCUSSION TITLE
USB PHY error/Invalid Packet ID in full speed capture?

### DISCUSSION THREAD

**Discussion by dougg3** [ataradov/usb-sniffer#11] [Q&A] [answered]
Hey,

This project is awesome! I built one up today. It works great! I have run into one strange problem and was wondering if this is normal. Here is what I do:

- Start out with both USB-C ports plugged into two ports in my computer (one for target, one for host)
- Set the correct capture speed in Wireshark (4.0.6) and start the capture
- Plug a device into the USB-A port on the USB Sniffer

This works fine with all three speeds. I've captured USB traffic successfully at all speeds, and it looks correct. But...in full speed mode, I randomly see errors like this until I plug a device into the USB-A port:

```
Syslog: USB PHY error
Invalid Packet ID (0xff)
```

The packet ID doesn't always say 0xff, but that's the most common one. Sometimes it happens fast, sometimes it only happens every few seconds.

As soon as I plug in a full-speed device, it stops doing that and everything works fine. This only seems to happen to me in full speed mode. I don't see this problem with high speed or low speed. Is this normal? It's also very possible I made some kind of mistake assembling everything. This is the first time I've soldered a QFN package before. All the connections look good under the microscope, but this error is making me doubt myself.

![Screenshot 2023-07-08 20-42-00](https://github.com/ataradov/usb-sniffer/assets/1565758/7d578deb-fa0c-48d7-9bce-261a035d3827)

Thanks for any insight! Really nice job on this project!

**Comment by ataradov** [comment, marked as answer]
With no device connected the line is essentially floating, so it would be picking up noise from the cable, which is interpreted by the PHY as valid start of frame, but then it fails to receive it correctly, so PHY Error is indicated. The characteristics of the noise are such that they match FS sync sequence, so other modes are not affected.

I have seen "noise" frames with no device plugged in when I initially built the device. In my case I swapped the PHY while investigating and it seemed to solve the issue. I'm not sure if there is variability in the PHYs, or something gets damaged during soldering or handling, or something else entirely. I did not really expect it to help, but it did in my case. 

There is nothing wrong with your assembly, it would not work at all otherwise. You can try to change the PHY and see if it does anything. Changing the cables for shorter ones or USB ports on the host may also help.

**Comment by ataradov** [comment]
Or you can just ignore those frames, they should not affect operation when the device is plugged in and you will get some noise line state messages from the contact bounce anyway. It is not possible to fully get rid of all the noise.

**Comment by dougg3** [comment]
Thank you for your quick response! I have parts to build 2 more, so I'll see if the other ones do it too. I've only tested one full speed device and cable, so maybe it's something about the cable I'm testing with. Yeah, I can definitely just ignore them -- everything is fine as soon as the device is plugged in.

**Comment by ataradov** [comment]
At the time I had a quick look with the scope on the lines and I did not see any nose, which is what made me suspect the PHY or soldering. I had another board with the same overall design, but with minor changes (though hole LEDs and stuff like that). This board was assembled a long time ago and I've never seen that issue on it, so I tried to both on the same exact cable and port setup and only new one showed the noise frames. So, I changed the PHY because I had no other thoughts and it solved the issue.

The only thing I payed attention to is to not apply excessive heat for a longer than necessary time. But I'm not sure if that has any effect, but this was the only thing I could control to some extent.

**Comment by simonboydfoley** [comment]
I had a similar issue ... turned out to be a disconnected pin on the PHY Chip 3343-CP   _"SPK_R"_.  Correction It was Pin "DATA7"
Originally was using Solder Paste and a hot plate. When I drag soldered the affected pin it all started to work :-)

**Comment by ataradov** [comment]
USB 3343 does not have SPK_R pin. There is a data pin where SPK_R is on other devices in the series.

**Reply by simonboydfoley** [reply]
Yup sorry I was looking at the wrong data sheet. It was Pin "DATA7"  on the PCB it was labelled as  6 T_USB_D2
